notifications/reactive/src/test/java/org/trellisldp/reactive/ReactiveEventServiceTest.java

```diff
@@ -0,0 +1,117 @@
+
+import java.time.Instant;
+
+import org.apache.commons.rdf.api.RDF;
+import org.apache.commons.rdf.simple.SimpleRDF;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.trellisldp.api.*;
+import org.trellisldp.event.DefaultActivityStreamService;
+import org.trellisldp.vocabulary.AS;
+import org.trellisldp.vocabulary.LDP;
+import org.trellisldp.vocabulary.Trellis;
+
+public class ReactiveEventServiceTest {
+
+    private static final RDF rdf = new SimpleRDF();
+    private static WeldContainer container;
+
+    private final Instant time = now();
+
+    @Mock
+    private Event mockEvent;
+
+    @BeforeAll
+    public static void initialize() {
+        // Setup Weld
+        final Weld weld = new Weld();
+        weld.addBeanClass(MediatorFactory.class);
+        weld.addBeanClass(MediatorManager.class);
+        weld.addBeanClass(InternalChannelRegistry.class);
+        weld.addBeanClass(ConfiguredChannelFactory.class);
+
+        weld.addExtension(new ReactiveMessagingExtension());
+
+        weld.addBeanClass(TestCollector.class);
+        weld.addBeanClass(ReactiveEventService.class);
+        weld.addBeanClass(DefaultActivityStreamService.class);
+        weld.addBeanClass(ReactiveEventService.class);
+        weld.disableDiscovery();
+
+        container = weld.initialize();
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        if (container != null) {
+            container.shutdown();
+        }
+    }
+
+    @BeforeEach
+    public void setUp() {
+        initMocks(this);
+
+        when(mockEvent.getTarget()).thenReturn(of(rdf.createIRI(TRELLIS_DATA_PREFIX + "resource")));
+        when(mockEvent.getAgents()).thenReturn(singleton(Trellis.AdministratorAgent));
+        when(mockEvent.getIdentifier()).thenReturn(rdf.createIRI("urn:test"));
+        when(mockEvent.getCreated()).thenReturn(time);
+        when(mockEvent.getTypes()).thenReturn(singleton(AS.Update));
+        when(mockEvent.getTargetTypes()).thenReturn(singleton(LDP.RDFSource));
+        when(mockEvent.getInbox()).thenReturn(empty());
+    }
+
+    @Test
+    public void testReactiveStream() {
+        final TestCollector collector = getInstance(TestCollector.class);
+        final ReactiveEventService service = getInstance(ReactiveEventService.class);
+
+        service.emit(mockEvent);
+        await().atMost(FIVE_SECONDS).until(() -> collector.getResults().size() == 1);
+        service.emit(mockEvent);
+        await().atMost(FIVE_SECONDS).until(() -> collector.getResults().size() == 2);
+        service.emit(mockEvent);
+        service.emit(mockEvent);
+        await().atMost(FIVE_SECONDS).until(() -> collector.getResults().size() == 4);
+    }
+
+    private <T> T getInstance(final Class<T> type) {
```

Comment:
If the idea is to get properly set-up beans from the container, can we not just inject them? I'm not at all familiar with Weld's test gear so maybe not?

Reply:
I'm also not super familiar with Weld's test gear, but I'll see if there is something easy that will do that injection for us.

Reply:
https://github.com/weld/weld-junit/tree/master/junit5

Reply:
That last commit should address this. It also makes the test code much more compact.